src/prefect/utilities/filesystem.py

```diff
@@ -117,3 +120,45 @@ def relative_path_to_current_platform(path_str: str) -> Path:
     """
 
     return Path(PureWindowsPath(path_str).as_posix())
+
+
+def _copytree_37(src, dst, original_src, symlinks=False, ignore=None):
+    """
+    Replicates the behavior of `shutil.copytree(src=src, dst=dst, ignore=ignore, dirs_exist_ok=True)`
+    in a python 3.7 compatible manner.
+
+    Source for the logic: Cyrille Pontvieux at https://stackoverflow.com/a/22331852
+    """
+    if not os.path.exists(dst):
+        os.makedirs(dst)
+        shutil.copystat(src, dst)
+    elements = os.listdir(src)
+
+    if ignore:
+        exclude = ignore(src, elements)
+        elements = [el for el in elements if el not in exclude]
+
+    for item in elements:
+        source_path = os.path.join(src, item)
+        destination_path = os.path.join(dst, item)
+        if symlinks and os.path.islink(source_path):
+            if os.path.lexists(destination_path):
+                os.remove(destination_path)
+            os.symlink(os.readlink(source_path), destination_path)
+            try:
+                st = os.lstat(source_path)
+                mode = stat.S_IMODE(st.st_mode)
+                os.lchmod(destination_path, mode)
+            except:
+                pass  # lchmod not available
+        elif os.path.isdir(source_path):
+            _copytree_37(source_path, destination_path, original_src, symlinks, ignore)
+        else:
+            shutil.copy2(source_path, destination_path)
+
+
+def prefect_copytree(src, dst, ignore=None):
+    if sys.version_info < (3, 8):
+        _copytree_37(src=src, dst=dst, original_src=src, ignore=ignore)
+    else:
+        shutil.copytree(src=src, dst=dst, ignore=ignore, dirs_exist_ok=True)
```

Comment:
I think `prefect.utilities.compat.copytree` makes more sense to me for all of this. I'd do something like

if if sys.version_info < (3, 8):
   def copytree(...):
      ..
else:
   from shutil import copytree
```

No reason to define this function if not running on 3.7.